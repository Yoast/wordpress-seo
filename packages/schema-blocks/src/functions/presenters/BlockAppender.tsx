import { createElement } from "@wordpress/element";
import { Inserter } from "@wordpress/block-editor";

/**
 * Properties for the BlockAppender.
 */
export interface BlockAppenderProps {
	clientId: string;
	label: string;
}

/**
 * Block Appender for adding a new block to an innerblocks block.
 *
 * @param clientId The id of the parent block in which a block should be added when using the appender.
 * @param label A string
 *
 * @constructor
 */
export default function BlockAppender( { clientId, label }: BlockAppenderProps ): React.ReactElement {
	return (
		<div
			data-root-client-id={ clientId }
			className="wp-block block-editor-default-block-appender"
		>
			<p className="block-editor-default-block-appender__content yoast-schema-block-appender__content">
				{ label }
			</p>
			<Inserter
				rootClientId={ clientId }
				position="bottom right"
			/>
		</div>
	);
}
