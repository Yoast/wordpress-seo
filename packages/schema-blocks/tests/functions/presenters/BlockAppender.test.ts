import * as renderer from "react-test-renderer";
import BlockAppender from "../../../src/functions/presenters/BlockAppender";

jest.mock( "@wordpress/block-editor", () =>  ( {
	Inserter: () => "BlockInserter",
} ) );

describe( "The BlockAppender", () => {
	it( "renders correctly", () => {
		const tree = renderer
			.create( BlockAppender( { clientId: "1234-abcd", label: "Add a block to your job posting..." } ) )
			.toJSON();

		expect( tree ).toMatchSnapshot();
	} );
} );
