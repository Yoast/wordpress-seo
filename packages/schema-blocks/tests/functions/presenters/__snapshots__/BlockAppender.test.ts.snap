// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The BlockAppender renders correctly 1`] = `
<div
  className="wp-block block-editor-default-block-appender"
  data-root-client-id="1234-abcd"
>
  <p
    className="block-editor-default-block-appender__content yoast-schema-block-appender__content"
  >
    Add a block to your job posting...
  </p>
  BlockInserter
</div>
`;
